src/run_common.cpp

```diff
@@ -36,6 +36,12 @@ void setup_common_run_options(
             "t is the current logical time and d is the amount of logical "
             "time that has passed since the start of the simulation.  "
             "t and d are floating-point numbers while n is an integer.")
+        ("worker-threads",
+            boost::program_options::value<unsigned int>()->value_name("worker-threads")->implicit_value(0),
+            "Enables spawning worker-threads to parallelize the work load. "
+            "If not specified, the number of worker-threads is set to the "
+            "number of the system hardware cores -1. Worker-threads comes "
+            "in addition to the application thread.")
```

Comment:
Should check if we don't want to create any threads when the option is not given (related to the above comments).

Reply:
This would by default run the simulation on the main thread. Implicit value could be -1 and converted to std::nullopt

Reply:
Note that `unsigned int` can't be -1, so this will wrap around. I suggest using `int` and converting to `unsigned` later.

Reply:
Fixed as suggested.